packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -313,6 +318,37 @@ export default class DatagridController {
         };
     }
 
+    getSelectedRows () {
+        return this.selectedRows.reduce((result, selected, index) => {
+            if (selected) {
+                result.push(this.displayedRows[index]);
+            }
+            return result;
+        }, []);
+    }
+
+    onSelectRow (index, selected) {
+        const rowCount = this.displayedRows.length;
+        this.selectedRows[index] = selected;
+        const selectedRowsCount = this.getSelectedRows().length;
+
+        if (selectedRowsCount === rowCount) {
+            this.selectAllRows = true;
+        } else if (selectedRowsCount === 0) {
+            this.selectAllRows = false;
+        } else {
+            this.selectAllRows = null;
```

Comment:
Is this the global comportement for tri-select box ? `null` is `some items are selected, some are not`? Because, if it's not already the selected behavior, this would require a something else than a boolean then

Reply:
yep checkbox component currently use "null"